src/sentry/static/sentry/app/views/settings/incidentRules/ruleConditionsForm.tsx

```diff
@@ -123,9 +124,15 @@ class RuleConditionsForm extends React.PureComponent<Props, State> {
                 useFormWrapper={false}
                 organization={organization}
                 onChange={onChange}
-                onBlur={onBlur}
                 onKeyDown={onKeyDown}
-                onSearch={query => onChange(query, {})}
+                onBlur={query => {
+                  onBlur(query);
+                  this.props.onFilterUpdate(query);
+                }}
+                onSearch={query => {
+                  this.props.onFilterUpdate(query);
+                  onChange(query, {});
```

Comment:
Should the order that the onFilterUpdate and onChange/onBlur be the same?

Reply:
Order doesn't matter, but updated to be consistent